guides/micronaut-http-client/groovy/src/test/groovy/example/micronaut/GithubControllerSpec.groovy

```diff
@@ -1,60 +1,77 @@
-    HttpClient client // <2>
+    @Ignore
+    void "verify GithubReleases can Be fetched With Low Level Http Client"() {
+        given:
+        EmbeddedServer github = ApplicationContext.run(EmbeddedServer,
+                Collections.singletonMap("spec.name", "GithubControllerTest")) // <1>
+        String url = "http://localhost:${github.port}"
+        EmbeddedServer embeddedServer = ApplicationContext.run(EmbeddedServer,
+                Collections.singletonMap("micronaut.http.services.github.url", url)) // <2>
+        HttpClient httpClient = embeddedServer.getApplicationContext()
+                .createBean(HttpClient, embeddedServer.getURL())
+        BlockingHttpClient client = httpClient.toBlocking()
 
-    @Shared
-    Pattern MICRONAUT_RELEASE = Pattern.compile("[Micronaut|Micronaut Framework] [0-9].[0-9].[0-9]([0-9])?( (RC|M)[0-9])?");
-
-
-    void 'verify github releases can be fetched with low level HttpClient'() {
         when:
-        HttpRequest request = HttpRequest.GET('/github/releases-lowlevel')
-
-        HttpResponse<List<GithubRelease>> rsp = client.toBlocking().exchange(request, // <3>
+        HttpRequest<?> request = HttpRequest.GET("/github/releases-lowlevel")
+        HttpResponse<List<GithubRelease>> rsp = client.exchange(request, // <3>
                 Argument.listOf(GithubRelease)) // <4>
 
-        then: 'the endpoint can be accessed'
-        rsp.status == HttpStatus.OK // <5>
-        rsp.body() // <6>
+        then:
+        HttpStatus.OK == rsp.getStatus()   // <5>
 
         when:
-        List<GithubRelease> releases = rsp.body()
+        List<GithubRelease> releases = rsp.body() // <6>
 
         then:
         releases
         releases.stream()
                 .map(GithubRelease::getName)
                 .allMatch(name -> MICRONAUT_RELEASE.matcher(name)
                         .find())
-    }
 
-    void 'verify github releases can be fetched with compile-time autogenerated @Client'() {
-        when:
-        HttpRequest request = HttpRequest.GET('/github/releases-lowlevel')
+        cleanup:
+        httpClient.close()
+        embeddedServer.close()
+        github.close()
+    }
 
-        List<GithubRelease> githubReleases = client.toBlocking().retrieve(request, Argument.listOf(GithubRelease)) // <7>
+    @Requires(property = "spec.name", value = "GithubControllerTest") // <1>
+    @Controller
+    static class GithubReleases {
+        private final ResourceLoader resourceLoader
+        GithubReleases(ResourceLoader resourceLoader) {
+            this.resourceLoader = resourceLoader
+        }
 
-        then:
-        githubReleases.stream()
-                .map(GithubRelease::getName)
-                .allMatch(name -> MICRONAUT_RELEASE.matcher(name)
-                        .find())
+        @Consumes(value = ["application/vnd.github.v3+json", "application/json"])
+        @Get("/repos/micronaut-projects/micronaut-core/releases") // <3>
+        String coreReleases() {
```

Comment:
I don't know if it is Apache Groovy or a bug in Micronaut Framework but if I set it does not work. 

```groovy
Optional<InputStream> coreReleases() {
            resourceLoader.getResourceAsStream("releases.json")
}

Reply:
I've seen cases where Groovy didn't resolve resources with the same path. You could try to use an absolute path to the resource instead.

Reply:
Please, note that `resourceLoader.getResourceAsStream("releases.json")` seems to work. In order to make it work, in Apache Groovy I have to unwrap the optional and get the string of input stream with Groovy's `.text`. Thus, I think it is not an issue with the path. It seems the issue is returning an `Optional<InputStream>` with Apache Groovy.

Reply:
Ah sorry I missed it wasn't a `ClassLoader` here but a resource loader.